gprofiler/log.py

```diff
@@ -125,6 +127,8 @@ def emit(self, record: LogRecord) -> None:
             return
 
         self._logs.append(self._make_dict_record(record))
+        # trim logs to last N entries
+        self._logs[: -self.MAX_BUFFERED_RECORDS] = []
```

Comment:
We should be aware that a trimming happened, at least once. Perhaps add a single log that the logs were trimmed, set a bool flag in the logger that the trimming message was logged, and reset the flag on a successful logs upload?

Reply:
Good call. I'll add such log. Logically, as long as trimming has happened, there'll always be one such log in the buffer (because once one gets trimmed, a new one is added).

Reply:
I don't think your logic works, seems to me like it would remove the new "truncated" log